haystack/file_converter/pdf.py

```diff
@@ -130,7 +130,7 @@ def _read_pdf(self, file_path: Path, layout: bool, encoding: str) -> List[str]:
             command = ["pdftotext", "-enc", encoding, "-layout", str(file_path), "-"]
```

Comment:
I guess we possibly do not need the `Optional` type? The `command` may not work with a `None` value for `encoding`.

Reply:
Actually it is coming from `convert` where encoding param is `Optional[str]` we need to change there otherwise mypy will complain. I do not wanted to bring so many changes hence only corrected this private function signature